packages/medusa-payment-stripe/src/core/stripe-base.ts

```diff
@@ -0,0 +1,327 @@
+    const options: PaymentIntentOptions = {}
+
+    if (this?.paymentIntentOptions?.capture_method) {
+      options.capture_method = this.paymentIntentOptions.capture_method
+    }
+
+    if (this?.paymentIntentOptions?.setup_future_usage) {
+      options.setup_future_usage = this.paymentIntentOptions.setup_future_usage
+    }
+
+    if (this?.paymentIntentOptions?.payment_method_types) {
+      options.payment_method_types =
+        this.paymentIntentOptions.payment_method_types
+    }
+
+    return options
+  }
+
+  async getPaymentStatus(
+    paymentSessionData: Record<string, unknown>
+  ): Promise<PaymentSessionStatus> {
+    const id = paymentSessionData.id as string
+    const paymentIntent = await this.stripe_.paymentIntents.retrieve(id)
+
+    switch (paymentIntent.status) {
+      case "requires_payment_method":
+      case "requires_confirmation":
+      case "processing":
+        return PaymentSessionStatus.PENDING
+      case "requires_action":
+        return PaymentSessionStatus.REQUIRES_MORE
+      case "canceled":
+        return PaymentSessionStatus.CANCELED
+      case "requires_capture":
+      case "succeeded":
+        return PaymentSessionStatus.AUTHORIZED
+      default:
+        return PaymentSessionStatus.PENDING
+    }
+  }
+
+  async initiatePayment(
+    context: PaymentProcessorContext
+  ): Promise<PaymentProcessorError | PaymentProcessorSessionResponse> {
+    const intentRequestData = this.getPaymentIntentOptions()
+    const {
+      email,
+      context: cart_context,
+      currency_code,
+      amount,
+      resource_id,
+      customer,
+    } = context
+
+    const description = (cart_context.payment_description ??
+      this.options_?.payment_description) as string
+
+    const intentRequest: Stripe.PaymentIntentCreateParams = {
+      description,
+      amount: Math.round(amount),
+      currency: currency_code,
+      metadata: { resource_id },
+      capture_method: this.options_.capture ? "automatic" : "manual",
+      ...intentRequestData,
+    }
+
+    if (this.options_?.automatic_payment_methods) {
+      intentRequest.automatic_payment_methods = { enabled: true }
+    }
+
+    if (customer?.metadata?.stripe_id) {
+      intentRequest.customer = customer.metadata.stripe_id as string
+    } else {
+      let stripeCustomer
+      try {
+        stripeCustomer = await this.stripe_.customers.create({
+          email,
+        })
+      } catch (e) {
+        return this.buildError(
+          "An error occurred in InitiatePayment during the creation of the stripe customer",
+          e
+        )
+      }
+
+      intentRequest.customer = stripeCustomer.id
+    }
+
+    let session_data
+    try {
+      session_data = (await this.stripe_.paymentIntents.create(
+        intentRequest
+      )) as unknown as Record<string, unknown>
+    } catch (e) {
+      return this.buildError(
+        "An error occurred in InitiatePayment during the creation of the stripe payment intent",
```

Comment:
todo: Simplify error message
```suggestion
          "An error occurred in initiatePayment when creating a Stripe customer",
          e
        )
      }

      intentRequest.customer = stripeCustomer.id
    }

    let session_data
    try {
      session_data = (await this.stripe_.paymentIntents.create(
        intentRequest
      )) as unknown as Record<string, unknown>
    } catch (e) {
      return this.buildError(
        "An error occurred in initiatePayment",
```